### ISSUE TITLE
warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)

### ISSUE THREAD

**Issue by CarlosCarPer** [pytorch/pytorch#40091]
I'm working on braincode master
([https://github.com/braindecode/braindecode](url)),
concretly on the plot_bcic_iv_2a_moabb_trial.py ([https://braindecode.org/auto_examples/plot_bcic_iv_2a_moabb_trial.html](url))

When I run it all seems well, but at the end I get this:

  epoch    train_accuracy    train_loss    valid_accuracy    valid_loss     dur
-------  ----------------  ------------  ----------------  ------------  ------
      1            0.2500        1.5764            0.2500        5.5867  9.4235
C:\Users\Carci\anaconda3\lib\site-packages\torch\optim\lr_scheduler.py:143: UserWarning: The epoch parameter in `scheduler.step()` was not necessary and is being deprecated where possible. Please use `scheduler.step()` to step the scheduler. During the deprecation, if epoch is different from None, the closed form is used instead of the new chainable form, where available. Please open an issue if you are unable to replicate your use case: https://github.com/pytorch/pytorch/issues/new/choose.
  warnings.warn(EPOCH_DEPRECATION_WARNING, UserWarning)
      2            0.2604        1.1770            0.2500        6.7523  8.5649
      3            0.2604        1.0862            0.2535        5.6064  8.5578
      4            0.2812        0.9711            0.2569        4.2633  8.0839

I'm very new to code on Python (this is for my final grade work) and I would thank help to fix or remove this warning.

cc @vincentqb

**Comment by malfet** [comment]
This most likely comes from https://github.com/skorch-dev/skorch/blob/master/skorch/callbacks/lr_scheduler.py#L170

**Comment by CarlosCarPer** [comment]
So I should find the lr_scheduler.py and modify or remove the line?

I'm trying to find the file, but there are like 5 files with that filename. The first I look was the one on the directory the warning returned mem but it's not the same code that you linked me. Maybe I should replace it? Or reinstall the skorch library?

Sorry and thank you for the help.

**Comment by vincentqb** [comment]
Thanks for reporting this. The learning rate values should still be correct, even with the warning. However, the issue should be opened with skorch, so that they update their library with the new convention that came with #26423. I will close this issue, but please feel free to re-open if needed.
